### PR TITLE
UIBULKED-20: Bulk edit confirmation modal

### PR DESCRIPTION
**Pull request by UladzislauKutarkin**
## Purpose

https://issues.folio.org/browse/UIBULKED-20
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by uladislausamets — APPROVED** [review]

File: src/components/BulkEditStartModal/BulkEditStartModal.js [hunk 1]
```diff
@@ -40,15 +50,47 @@ const BulkEditStartModal = ({
     setDropZoneActive(false);
   };
 
+  const uploadFileFlow = async (fileToUpload) => {
+    setFileName(fileToUpload.name);
+    setDropZoneActive(false);
+
+    try {
+      const { id } = await requestJobId({ recordIdentifier: 'BARCODE', editType: BULK_EDIT_UPDATE });
```
Comment (vashjs) [inline src/components/BulkEditStartModal/BulkEditStartModal.js:58]: may we move recordIdentifiers in constants?

**Review by vashjs — APPROVED** [review]

File: src/components/BulkEditStartModal/BulkEditStartModal.js [hunk 2]
```diff
@@ -40,15 +50,47 @@ const BulkEditStartModal = ({
...
+        message: errorMessage,
+        type: 'error',
+      });
+    }
+  };
+
+  const handleDrop = (acceptedFiles) => {
+    const fileToUpload = acceptedFiles[0];
+
+    uploadFileFlow(fileToUpload);
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/components/BulkEditStartModal/BulkEditStartModal.js:78]: Fixed.

File: src/components/BulkEditStartModal/BulkEditStartModal.js (hunk 1 above)
Comment (UladzislauKutarkin) [reply src/components/BulkEditStartModal/BulkEditStartModal.js:58]: I have already moved it. Thanks for your suggestion.

File: test/jest/utils/fileUpload.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  Object.defineProperty(file, 'size', {
+    get() {
+      return size;
+    },
+  });
+  return file;
+}
+
+export function flushPromises(container) {
+  return new Promise(resolve => setImmediate(() => {
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply test/jest/utils/fileUpload.js:43]: Fixed it already

File: src/components/BulkEditStartModal/BulkEditStartModal.js [hunk 4]
```diff
@@ -40,15 +50,47 @@ const BulkEditStartModal = ({
...
+      const { id } = await requestJobId({ recordIdentifier: 'BARCODE', editType: BULK_EDIT_UPDATE });
+
+      const data = await fileUpload({ id, fileToUpload });
+
+      await setUpdatedId(id);
+
+      setCountOfRecords(data);
+
+      setConformationButton(false);
+    } catch ({ message }) {
```
[the remark this answers is not in the document]
Comment (UladzislauKutarkin) [reply src/components/BulkEditStartModal/BulkEditStartModal.js]: Removed unused variable.

**Pull request merged**